File: src/org/frankversnel/nl/poortjes/game/component/Player.java
package org.frankversnel.nl.poortjes.game.component;


import org.frankversnel.nl.poortjes.GameObject;
import org.frankversnel.nl.poortjes.collision.Collision;
import org.frankversnel.nl.poortjes.collision.CollisionListener;
import org.frankversnel.nl.poortjes.collision.component.Collidable;
import org.frankversnel.nl.poortjes.component.Component;

public class Player extends Component implements CollisionListener {
	public Player(GameObject gameObject, Collidable collidable) {
		super(gameObject);
		
		collidable.addListener(this);
	}

	@Override
	public void collision(Collision collision) {
		Candy candy = collision.getCollidedWith().getComponent(Candy.class);
		if (candy != null) {
			collision.getCollidedWith().destroy();
		}
		
	}

}

File: src/main/scala/org/frankversnel/poortjes/game/gameobjects/player.scala
package org.frankversnel.poortjes.game.gameobjects

import org.frankversnel.poortjes._
import org.frankversnel.poortjes.rendering.DrawableShape
import org.frankversnel.poortjes.resource_loading.ResourceLoader
import org.frankversnel.poortjes.resource_loading.ResourceId
import org.frankversnel.poortjes.collision._
import org.frankversnel.poortjes.moving._
import org.frankversnel.poortjes.util.GameConfiguration

abstract class Player(val resourceLoader: ResourceLoader) extends DrawableShape
		with CollidableRectangle with Moveable with Speed {
	//speed
	val distanceInMs = GameConfiguration.getProperty("player_move_speed").toFloat
	val rotationInMs = GameConfiguration.getProperty("player_rotation_speed").toFloat

	var dimension = DimensionValue().width(9).height(13)

	val shape: ResourceId

	def onCollision(collider: Collidable, update: Update) {
		if(collider.isInstanceOf[Candy]) {
			val score = update.gameObjects.filter(_.isInstanceOf[Score]).headOption
			if(score.isDefined) {
				score.get.asInstanceOf[Score].incrementMultiplier
			}

			collider.destroy
		}
	}
}
